### ISSUE TITLE
Github repository containers  in codespace

### ISSUE THREAD

**Issue by bhack** [microsoft/vscode-dev-containers#713]
Please make [repository-containers](https://github.com/microsoft/vscode-dev-containers/tree/master/repository-containers/github.com) contributions automatically available in codespaces.

**Comment by bhack** [comment]
Can we figure out this before Codespaces GA https://github.com/github/roadmap/issues/55?

**Comment by bhack** [comment]
@Chuxel Can we rely on `.devcontainers` hosted here or we need to move in our repository for Codespaces?

**Comment by Chuxel** [comment]
@bhack I talked with @chrmarti and there's currently a technical issue preventing this from working. The determination as to whether to fall back on the default Codespaces image (codespaces-linux) is made based on the presence or absence of devcontainer.json in Codespaces. (That logic does not exist for Remote - Containers.) That is preventing the repository-containers logic from kicking in despite the fact that there's a unified implementation between the two. Some additional work will be required to make this work as expected. That said, we're finding most people are not using repository containers so we'll need too see how difficult this would be to resolve.

(FYI @jkeech @joshspicer @edgonmsft)

**Comment by bhack** [comment]
Thanks, let me know cause if It will not be solved for GA we will host these file in the upstream repos.

**Comment by jkeech** [comment]
I would propose deprecating the repository configs hosted in this repo. The main purpose was to allow configs when upstream repos did not want to accept those contributions to enable Remote - Containers. But now that Codespaces is a GitHub feature, I'd expect most maintainers to be willing to add support for Codespaces if someone tries to contribute it.

Keeping the pattern simple and having the devcontainers live in the upstream repos will make it easier for users to understand how things are setup. Anyone who looks at a repo can tell whether or not a custom image/config is being used for Codespaces, and if so, what it contains.

**Comment by Chuxel** [comment]
@jkeech Yeah, @bhack actually has a counterpoint here for tensor flow. Not all OSS maintainers are going to be willing to bring in service or tool specific config on principle - particularly for big repositories with lots of different tools being used by contributors. (I can fill you in offline.) We also have the example of [Angluar](https://github.com/angular/angular/tree/master/.devcontainer) which does have a devcontainer folder, but is disabled by default.

Remote - Containers also supports the idea of personal repository containers for a more nuanced use case that's also been useful and was added based on customer feedback. I am confident Codespaces will want this over time along with org defined definitions and overrides.

**Comment by bhack** [comment]
The real feature, if we will host the files in the upstream repos, is to let the users to use overlays https://github.com/microsoft/vscode-remote-release/issues/3279 for their customization without creating too much conflicts in forks.
